packages/gatsby-source-contentful/src/download-contentful-assets.js

```diff
@@ -45,6 +45,9 @@ const downloadContentfulAssets = async gatsbyFunctions => {
       const { contentful_id: id, node_locale: locale } = node
       const remoteDataCacheKey = `contentful-asset-${id}-${locale}`
       const cacheRemoteData = await cache.get(remoteDataCacheKey)
+      if (!node.file) {
+        return Promise.resolve()
```

Comment:
I'm not 100% sure if `node.name` exists but it should be the name of the asset. Something along the lines of this would be nice
```suggestion
        reporter.warn(`The asset ${node.name} contains no file.`)
        return Promise.resolve()
```

Reply:
I would add the id as well (it helps find the asset in contentful):
```
reporter.warn(`The asset ${node.name} with id: ${id} contains no file.`)
```

Reply:
@wardpeet checking here, I think it will be safer to just show the id - because even name could not be present in asset yet.